pinot-common/src/main/java/org/apache/pinot/common/config/RoutingConfig.java

```diff
@@ -25,25 +25,48 @@
 
 
 public class RoutingConfig extends BaseJsonConfig {
+  public static final String PARTITION_AWARE_SEGMENT_PRUNER_TYPE = "partitionAware";
```

Comment:
IMO `partition segment pruner` seems to be better name? up to you.

Reply:
Done